### PR TITLE
Events page

### PR DESCRIPTION
**Pull request by Doraemon012**
## Added the **Events Page**
This PR adds an `Events Page` to the website.

### Changes made:
- Added an Event List.
- Added images of various events.
- Created an EventCard component
- Created the Events Page which consists of the following
    -  Separate grids for **Main Events** and **Other Events**.
    - **Event cards** in the grids.
    - Each card displays the thumbnail, name, subtitle, a concise description, date, link to the website (if exists), and an image gallery (if images exist)

### Screen Recording:

https://github.com/BitByte-TPC/TPC-website/assets/116624667/43cf750e-4e4e-4ddc-9c28-e3b69ec7949e

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by aryan1113** [comment]
Add an href to the CodeRumble website too https://coderumble.vercel.app/

**Comment by Doraemon012** [comment]
@aryan1113 Done !

File: frontend/src/components/Cards/EventCard.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  createStyles({
+    linkContainer: {
+      "&:hover": {
+        background: "#03d7e03b",
+        color: "cyan",
+      },
+      marginTop: ".8rem",
+      width: "5rem",
+      paddingInline: ".6rem",
+      background: "#afdbdd3b",
```
Comment (Aksh-Bansal-dev) [inline frontend/src/components/Cards/EventCard.tsx]: Avoid hard-coding values whenever possible. You can use [CSS variables](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties).

**Review by Aksh-Bansal-dev — COMMENTED** [review]
Please compress images over 500KB.

Also, make sure that the UI is responsive.

**Review by Aksh-Bansal-dev — APPROVED** [review]
LGTM

**Pull request merged**